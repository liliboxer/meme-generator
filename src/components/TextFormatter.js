import React from 'react';
import PropTypes from 'prop-types';

function TextFormatter({ color, handleColorChange }) {
  return (
    <form>
      <input type="color" name="color" value={color} onChange={(event) => handleColorChange(event)}></input>
    </form>
  
  );

}

TextFormatter.propTypes = {
  color: PropTypes.string.isRequired,
  handleColorChange: PropTypes.func.isRequired
};

export default TextFormatter;
