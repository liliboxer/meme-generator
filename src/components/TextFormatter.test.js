import React from 'react';
import { shallow } from 'enzyme';
import TextFormatter from './TextFormatter';

describe('TextFormatter component', () => {
  it('renders TextFormatter', () => {
    const wrapper = shallow(<TextFormatter />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
