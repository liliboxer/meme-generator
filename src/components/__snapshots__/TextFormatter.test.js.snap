// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFormatter component renders TextFormatter 1`] = `
<form>
  <input
    name="color"
    onChange={[Function]}
    type="color"
  />
  <select
    name="font"
    onChange={[Function]}
  >
    <option
      key="Arial"
      value="Arial"
    >
      Arial
    </option>
    <option
      key="Helvetica"
      value="Helvetica"
    >
      Helvetica
    </option>
    <option
      key="Arial Black"
      value="Arial Black"
    >
      Arial Black
    </option>
    <option
      key="Comic Sans MS"
      value="Comic Sans MS"
    >
      Comic Sans MS
    </option>
    <option
      key="Impact"
      value="Impact"
    >
      Impact
    </option>
    <option
      key="Charcoal"
      value="Charcoal"
    >
      Charcoal
    </option>
    <option
      key="Lucida Sans Unicode"
      value="Lucida Sans Unicode"
    >
      Lucida Sans Unicode
    </option>
    <option
      key="Lucida Grande"
      value="Lucida Grande"
    >
      Lucida Grande
    </option>
    <option
      key="Courier New"
      value="Courier New"
    >
      Courier New
    </option>
    <option
      key="Lucida Console"
      value="Lucida Console"
    >
      Lucida Console
    </option>
  </select>
</form>
`;
